sklearn/decomposition/tests/test_pca.py

```diff
@@ -81,8 +81,9 @@ def test_whitening(solver, copy):
         whiten=True,
         copy=copy,
         svd_solver=solver,
-        random_state=0,
+        random_state=global_random_seed,
         iterated_power=7,
+        n_oversamples=13,
```

Comment:
After adjusting the hardcoded assertion thresholds in this test, a few tests with `randomized` solver still failed `assert_allclose(X_whitened, X_whitened2, rtol=5e-4)`. Increasing `n_oversamples` from default 10 to 13 solved this without adjusting the assertion tolerance.